File: bus-n-go-backend/src/main/kotlin/mk/ukim/finki/busngobackend/api/responses/RouteInstanceResponse.kt
package mk.ukim.finki.busngobackend.api.responses

import mk.ukim.finki.busngobackend.domain.entities.Avtobus
import mk.ukim.finki.busngobackend.domain.entities.Linija
import mk.ukim.finki.busngobackend.domain.entities.Pravec
import java.time.LocalDateTime

data class RouteInstanceResponse(
    val id: Long,
    val bus: Avtobus,
    val direction: Pravec,
    val line: Linija,
    val driver: UserResponse,
    val start: LocalDateTime,
    val end: LocalDateTime?,
)

File: bus-n-go-frontend/src/app/model/responses/RouteInstanceResponse.ts
import {Bus} from "../Bus";
import {Direction} from "../Direction";
import {Line} from "../Line";
import {UserResponse} from "./UserResponse";

export interface RouteInstanceResponse {
  bus: Bus,
  direction: Direction,
  line: Line,
  driver: UserResponse,
  start: Date,
  end: Date | undefined,
  id: number
}
